hadoop-hdds/tools/src/main/java/org/apache/hadoop/hdds/scm/cli/pipeline/CloseAllPipelinesSubcommand.java

```diff
@@ -0,0 +1,83 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ * <p>
+ * http://www.apache.org/licenses/LICENSE-2.0
+ * <p>
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.apache.hadoop.hdds.scm.cli.pipeline;
+
+import org.apache.hadoop.hdds.cli.HddsVersionProvider;
+import org.apache.hadoop.hdds.client.ReplicationFactor;
+import org.apache.hadoop.hdds.protocol.proto.HddsProtos;
+import org.apache.hadoop.hdds.scm.cli.ScmSubcommand;
+import org.apache.hadoop.hdds.scm.client.ScmClient;
+import org.apache.hadoop.hdds.scm.pipeline.Pipeline;
+import picocli.CommandLine;
+
+import java.io.IOException;
+import java.util.Optional;
+import java.util.function.Predicate;
+import java.util.stream.Stream;
+
+/**
+ * Handler of close pipeline command.
+ */
+@CommandLine.Command(
+    name = "closeAll",
```

Comment:
I suggest adding the new options to the existing `ozone admin pipeline close` command, instead of introducing a new command.  Existing parameter for pipeline ID can be changed to arity = 0..1.

Reply:
Arity means the number of times the parameter can/should be given to the command.

You want to use the new options (`--all` or the ones in the original patch) without any pipeline ID.  The command will complain about that:

```
Missing required parameter: '<pipelineId>'
```

`arity = "0..1"` makes the parameter optional.

https://picocli.info/#_optional_values